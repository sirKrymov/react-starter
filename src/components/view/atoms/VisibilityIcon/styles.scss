.visibility-icon {
  display: inline-flex;
  cursor: pointer;
}
