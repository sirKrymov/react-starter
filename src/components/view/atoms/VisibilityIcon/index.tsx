// @SETUP: Please don't forget to remove this component or current comment if component is used in project
import React, { ReactElement } from 'react';

import { SvgIcon } from '../SvgIcon';

import { eyeCrossedIcon, eyeIcon } from 'constants/icons';

import './styles.scss';

interface IProps {
  isVisible: boolean;
  onClick(): void;
  id: string;
}

export function VisibilityIcon({
  isVisible,
  onClick,
  id
}: IProps): ReactElement {
  return (
    <div className="visibility-icon" onClick={onClick} id={id}>
      <SvgIcon size="lg" src={isVisible ? eyeCrossedIcon : eyeIcon} />
    </div>
  );
}
