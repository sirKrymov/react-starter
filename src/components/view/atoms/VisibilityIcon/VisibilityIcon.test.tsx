import React from 'react';

import { shallow, ShallowWrapper } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import sinon from 'sinon';

import { VisibilityIcon } from '../VisibilityIcon';

describe('<VisibilityIcon> atom component', () => {
  const handleClick = sinon.spy();

  let shallowComponent: ShallowWrapper;

  const createComponent = ({ isVisible, onClick }: Record<string, any>) => {
    shallowComponent = shallow(
      <VisibilityIcon
        isVisible={isVisible}
        onClick={onClick}
        id="check-visibility-icon"
      />
    );
  };

  describe('WHEN the component loads', () => {
    beforeEach(() => {
      createComponent({ isVisible: false, onClick: handleClick });
    });

    it('THEN it should render without crashing', () => {
      expect(shallowComponent.html()).not.toHaveLength(0);
    });

    it('THEN it should match unvisible snapshot', () => {
      expect(shallowToJson(shallowComponent)).toMatchSnapshot();
    });
  });

  describe('WHEN the component loaded and have been clicked', () => {
    beforeEach(() => {
      createComponent({ isVisible: true, onClick: handleClick });
    });

    it('AND it should handle click  logic', () => {
      shallowComponent.find('div.visibility-icon').simulate('click');
      expect(handleClick).toHaveProperty('callCount', 1);
    });

    it('THEN it should match visible snapshot', () => {
      expect(shallowToJson(shallowComponent)).toMatchSnapshot();
    });
  });
});
