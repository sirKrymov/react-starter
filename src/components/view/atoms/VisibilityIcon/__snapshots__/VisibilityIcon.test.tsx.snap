// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<VisibilityIcon> atom component WHEN the component loaded and have been clicked THEN it should match visible snapshot 1`] = `
<div
  className="visibility-icon"
  id="check-visibility-icon"
  onClick={[Function]}
>
  <SvgIcon
    size="lg"
    src="eye_crossed.svg"
  />
</div>
`;

exports[`<VisibilityIcon> atom component WHEN the component loads THEN it should match unvisible snapshot 1`] = `
<div
  className="visibility-icon"
  id="check-visibility-icon"
  onClick={[Function]}
>
  <SvgIcon
    size="lg"
    src="eye.svg"
  />
</div>
`;
